src/layer/AbstractLayer.ts

```diff
@@ -30,7 +30,12 @@ export class AbstractLayer {
     this.legendUrl = legendUrl;
   }
 
-  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
+  @timeoutWrapper(2)
+  public async getMap(
+    params: GetMapParams,
+    api: ApiType,
+    requestsConfig?: RequestConfiguration,
```

Comment:
It is difficult to review this MR because so many (unrelated) things have changed. Maybe a good idea would be to implement just the timeout functionality, and leave cleanup tasks for another MR, so they can be properly checked too?